x/blob/ante/ante.go

```diff
@@ -24,11 +24,8 @@ func NewMinGasPFBDecorator(k BlobKeeper) MinGasPFBDecorator {
 // AnteHandle implemnts the AnteHandler interface. It checks to see
 // if the transaction contains a MsgPayForBlobs and if so, checks that
 // the transaction has allocated enough gas.
-//
-// TODO: We need to run this antehandler in process proposal
-// however this will be a breaking change.
 func (d MinGasPFBDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
-	if !ctx.IsCheckTx() {
+	if ctx.IsReCheckTx() {
```

Comment:
[non blocking][question] how does modifying this line make the antehandler run in ProcessProposal?

Reply:
ProcessProposal uses a context that has `isCheckTx` as false. Prior this would only run if the context was `CheckTx` now that this is removed it will run the handler in `ProcessProposal`. The reason for changing it to `IsReCheckTx` is that it doesn't make sense to re run this in `ReCheckTx` as nothing has changed

Reply:
> how does modifying this line make the antehandler run in ProcessProposal?

in #1984 we use the same antehandler for all things, which will then call the above handler as well if I'm not mistaken

Reply:
[no change needed][for my understanding] is there a downside to running this ante handler when `ctx.IsReCheckTx`? Is there a significant performance impact?

> The reason for changing it to IsReCheckTx is that it doesn't make sense to re run this in ReCheckTx as nothing has changed

To clarify, this ante handler is checking that the gas included the transaction exceeds the amount of gas expected to be consumed. Since it **does not** check that an account actually has the the amount of gas included in the transaction, this ante handler doesn't depend on state. Therefore the boolean result from this ante handler should be consistent before / after a new block is added even if that new block included a transaction that spent all of an account's funds.

Reply:
that's a good point @rootulp, but this ante handler works in tandem with the [`DeductFeeDecorator`](https://github.com/celestiaorg/cosmos-sdk/blob/dfc04d194ec23f99cd80e263a3cc75212e5cc1d0/x/auth/ante/fee.go#L104-L110) which runs before this one. So we actually first check if the account has enough funds, then we check that the fee is large enough for the PFB

Reply:
Does that mean the deduct fee decorator should be re-executed on ReCheckTx but the MinGasPFBDecorator in this PR shouldn't?

Reply:
perhpaps that's just prematurely optimal, but yeah

Reply:
> Is there a significant performance impact?

Probably not